www/scipylib/building/windows.rst

```diff
@@ -32,6 +32,172 @@ information than this document will provide.
+is running.
+
+First, install MSYS2 using `these instructions`_. Make sure to install the correct
+architecture for the SciPy that you want to build (eg. 32 or 64 bit), and make sure to
+install the MinGW, and git packages. Open a MSYS2 terminal and type the following:
+
+.. code:: shell
+   pacman -S --needed base-devel 
+       mingw-w64-i686-toolchain
+       mingw-w64-x86_64-toolchain
+       git subversion mercurial
+       mingw-w64-i686-cmake mingw-w64-x86_64-cmake
+       
+You should now have all of the tools required to build OpenBLAS. Now change directories
+to your favorite location with the following:
+ 
+.. code:: shell
+   cd /c/Users/MyUser/Downloads/
+   
+You don't necessarily need to build in that particular location, but it should be somewhere
+reasonable. Now clone the repository required to build OpenBLAS:
+
+.. code:: shell
+   git clone https://github.com/matthew-brett/build-openblas.git
+   cd build-openblas
+   git submodule update --init --recursive
+
+Now let's build OpenBLAS. To make sure that we're ready to build, type the following in
+the terminal:
+
+.. code:: shell
+   make
+   gfortran
+   gcc
+
+If any of these commands fail, you're not ready to build. Go back and make sure that
+MSYS2 is installed correctly and has the required packages enabled. Now, let's set
+some environment variables. In the MSYS2 terminal, type the following.
+
+.. code:: shell
+    export OPENBLAS_COMMIT=5f998ef
+    export OPENBLAS_ROOT="C:\\opt"
+    export BUILD_BITS=64
+
+Make sure that each variable makes sense. More specifically, make sure that the
+path that `OPENBLAS_ROOT` points to can be deleted. Make sure that the `OPENBLAS_COMMIT`
+points to the version that you want to build. Make sure that the architecture is
+correctly set. And after you've made sure of that, build OpenBLAS.
+
+.. code:: shell
+    .\build_openblas.sh
+
+Building OpenBLAS is extremely problematic and may fail if your system is not correctly
+configured. Your build may fail after a few hours and you may have to restart it after
+fixing an undocumented problem. OpenBLAS builds can also fail silently and produce
+and incorrect binary. Please, if you have any issues, `report them`_ so that we can save
+the next person's time.
+
+While you're waiting on OpenBLAS to finish building, go ahead and install `build tools`_
+from Microsoft, since these take a while to install and you'll need them later.
+
+After you've built OpenBLAS, there should be an `openblas.a` file somewhere on your system.
+If `OPENBLAS_ROOT` was set to `C:\\opt`, then you might see a line like this in the MSYS2
+terminal:
+
+.. code:: shell
+   Copying the static library to /c/opt/64/lib
+
+If you see that line, then you might have built OpenBLAS correctly, even if other failures
+occurred. Look in that folder for `openblas.a`. If the file isn't there, then poke around and
+try to find the file elsewhere. If you don't have that file, you'll probably need to find out
```

Comment:
@ilayn Build procedure has been reworked on focus on building OpenBLAS first and emphasizing the fact that the failures related to `lib.exe` that you experienced don't matter.

Reply:
@xoviat  No problem. I'm just adding the extra issues that comes from the `build_openblas.sh`. Note that the file name turns out as `libopenblas_5f998ef_gcc7_2_0.a` after this workflow (with postfix). So that also needs to be handled. Also I think you need extra empty lines after `..code: shell::` directives for them to be rendered.